guix: Add full installation and usage documentation

dongcarl opened the pull request:
Based on: #21462

Keeping the README in one file so that it's easy to search through. Will add more jumping links later so navigation is easier.

Current TODOs:
- [x] Shell installer option: prompt user to re-login for `/etc/profile.d` entry to be picked up
- [x] Binary tarball option: prompt user to create `/etc/profile.d` entry and re-login
- [x] Fanquake docker option: complete section
- [x] Arch Linux AUR option: prompt to start `guix-daemon-latest` unit after finishing "optional setup" section
- [x] Building from source option: Insert dependency tree diagram that I made
- [x] Building from source option: redo sectioning, kind of a mess right now
- [x] Optional setup: make clear which parts are only needed if building from source
- [x] Workaround 1 for GnuTLS: perhaps mention how to remove Guix build farm's key
- [x] Overall (after everything): Make the links work.

Note to self: wherever possible, tell user how to check that something is true rather than branching by installation option.

dongcarl committed:
guix: Check for a sane services database

On bare systems, it is possible to be lacking a services database. Check
for basic entries before attempting a build.

See the error message in the diff for more context.

78051301012 commented on contrib/guix/README.md:
```diff
@@ -1,5 +1,7 @@
 # Bootstrappable Bitcoin Core Builds
```
dostane btc

78051301012 reviewed (commented):
.

sipa commented on contrib/guix/README.md:
```diff
@@ -9,79 +11,835 @@ downloads.
+  `/var/guix/profiles/per-user/$USER/guix-profile`, and creates a symlink
+  targeting this directory at `$HOME/.guix-profile`
+
+Therefore, in order for these operations to effect your shell/desktop sessions
```
Nit: s/effect/affect/

sipa commented on contrib/guix/README.md:
````diff
@@ -9,79 +11,835 @@ downloads.
+Change the default list of substitute servers by starting `guix-daemon` with the
+`--substitute-urls` option:
+```
+guix <cmd> --substitute-urls="https://guix.carldong.io https://ci.guix.gnu.org"
````
I think you mean `guix-daemon` here.

TheCharlatan commented on contrib/guix/README.md:
```diff
@@ -9,79 +11,835 @@ downloads.
+Manual
+[here](https://guix.gnu.org/manual/en/html_node/Binary-Installation.html).
+Ignore the note about using the shell installer script since that does not
+apply. This is largely equivalent to maually performing what the shell installer
```
Nit: s/maually/manually/

TheCharlatan commented on contrib/guix/README.md:
````diff
@@ -9,79 +11,835 @@ downloads.
+sudo --login guix pull --commit=<particular-commit>
+```
+
+`guix pull` is quite a long process (espcially if you're using
````
Nit: s/espcially/especially/

TheCharlatan commented on contrib/guix/README.md:
````diff
@@ -9,79 +11,835 @@ downloads.
+export SUBSTITUTE_URLS='https://guix.carldong.io https://ci.guix.gnu.org'
+```
+
+## Option 2: Disabling subsitutes on an ad-hoc basis
````
Nit: s/subsitutes/substitutes/

fanquake commented:
@dongcarl can you rebase this and address the current batch of comments?

dongcarl commented:
Pushed 57684eda89...5ae2760f61
- Rebased over master
- Addressed https://github.com/bitcoin/bitcoin/pull/21711#pullrequestreview-638195828, https://github.com/bitcoin/bitcoin/pull/21711#pullrequestreview-638270471

MarcoFalke commented:
Shouldn't this also update `release-process.md`?

achow101 commented on contrib/guix/README.md:
```diff
@@ -9,79 +11,835 @@ downloads.
-## Setup
+# Installation and Setup
+
+At the time of writing, the latest Guix release is v1.2.0, which has a known
```
In 5ae2760f614c0872a2289574477543070243e236 "guix: Overhaul README"

Latest release is 1.3.0 which fixes the GnuTLS issue, so I think this paragraph is no longer needed.

achow101 reviewed (commented):
Concept ACK

More docs more good

jarolrod commented on contrib/guix/README.md:
```diff
@@ -9,79 +11,835 @@ downloads.
+4. Using a **distribution-maintained package** [⤓ skip to section]()
+   - Maintained by distribution's Guix package maintainer
+   - Normal difficulty (manual setup required)
+   - Works only on distributions with Guix packaged, at time of writing:
```
I don't think we should list the systems that have a guix package available. This will quickly diverge from reality. For example, a quick search shows openSuse should be added to the list: https://software.opensuse.org/package/guix

dongcarl replied:
I do think that we can give facts "at time of writing," sometimes it's better than nothing. However, in this particular case, perhaps we can simply link to https://repology.org/project/guix/versions ?

jarolrod commented on contrib/guix/README.md:
```diff
@@ -9,79 +11,835 @@ downloads.
+
+Follow the binary installation instructions step-by-step in the Guix Reference
+Manual
+[here](https://guix.gnu.org/manual/en/html_node/Binary-Installation.html).
```
This link contains the instructions that are found in section `Option 1: Using the official shell installer script`. 

Seems as though this section and the first (Option 1 & 2) can be condensed in some way.

For `Option 1`, i don't like that we are re-writing the instructions when we could just point to this section of the mentioned link:
![guix-install-script](https://user-images.githubusercontent.com/23396902/124206049-87a88680-dab0-11eb-865b-aa4d0bbc3f11.png)

dongcarl replied:
You're right, I'm not too happy with the current situation either... Will fix! Clearer = better

jarolrod reviewed (commented):
concept ACK

Couple of notes. I assume you are already working on an update for `1.3.0` so I will re-review when it's updated

dongcarl committed:
tree-wide: Rename gitian-keys to builder-keys

dongcarl committed:
guix: Adapt release-process.md to new Guix process

Also, clean up release-process.md

dongcarl committed:
guix-attest: Produce and sign normalized documents

That way we can easily combine the document and detached signature to
produce cleartext signature files for upload during the release process.

See subsequent commits which modify doc/release-process.md for more
details.

dongcarl commented:
The documentation is very close to being done, just some final polishing and making the links work. Please let me know if I'm missing any crucial details!

jarolrod commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,679 @@
+
+## Options 1 and 2: Using the official shell installer script or binary tarball
+
+The instructions for both using the offical shell installer script and using
```
```suggestion
The installation instructions for both the official shell installer script and the
```

jarolrod commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,679 @@
+## Options 1 and 2: Using the official shell installer script or binary tarball
+
+The instructions for both using the offical shell installer script and using
+binary tarballs are both found in the GNU Guix Manual's [Binary Installation
```
```suggestion
binary tarballs can be found in the GNU Guix Manual's [Binary Installation
```

jarolrod commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,679 @@
+
+## Option 4: Using a distribution-maintained package
+
+### Debian 11 ("Bullseye")/Ubuntu 21.04 ("Hirsute")
```
nit
```suggestion
### Debian 11 (Bullseye)/Ubuntu 21.04 (Hirsute)
```

jarolrod commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,679 @@
+- [Debian release history](https://www.debian.org/releases/)
+- [Ubuntu release history](https://ubuntu.com/about/release-cycle)
+
+### Arch Linux
```
no opinion, but we don't provide Arch Linux specific instructions anywhere else

dongcarl replied:
I think we do here: https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md#setup-and-build-example-arch-linux

jarolrod commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,679 @@
+### Debian 11 ("Bullseye")/Ubuntu 21.04 ("Hirsute")
+
+Guix v1.2.0 is available as a distribution package starting in [Debian
+11](https://packages.debian.org/bullseye/guix) ("bullseye") and [Ubuntu
```
If the nickname is already mentioned in the header along with the os version, we don't need to reiterate it here
```suggestion
11](https://packages.debian.org/bullseye/guix) and [Ubuntu
```

jarolrod commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,679 @@
+
+Guix v1.2.0 is available as a distribution package starting in [Debian
+11](https://packages.debian.org/bullseye/guix) ("bullseye") and [Ubuntu
+21.04](https://packages.ubuntu.com/hirsute/guix) (Hirsute Hippo).
```
and here

```suggestion
21.04](https://packages.ubuntu.com/hirsute/guix).
```

dongcarl committed:
guix: Overhaul README

- Added detailed Guix bootstrap/installation instructions

dongcarl commented:
This PR is ready for review as of 5016091dd8b827c74fe28212be43d3b22014c5b6

Thanks @jarolrod for the diligent reviews, I will address the nits after others agree that I haven't missed anything crucial!

achow101 commented on doc/release-process.md:
```diff
@@ -64,14 +60,14 @@ This will perform a few last-minute consistency checks in the build system files
 
-If you're using the automated script (found in [contrib/gitian-build.py](/contrib/gitian-build.py)), then at this point you should run it with the "--setup" command. Otherwise ignore this.
+Install Guix using one of the installation methods detailed in
+[contrib/guix/README.md](contrib/guix/README.md).
```
In 5016091dd8b827c74fe28212be43d3b22014c5b6 "guix: Adapt release-process.md to new Guix process"

I think this should refer to `contrib/guix/INSTALL.md`

dongcarl replied:
Fixed, thanks!

achow101 reviewed (commented):
ACK 5016091dd8b827c74fe28212be43d3b22014c5b6

fanquake reviewed (commented):
Concept ACK

Looks pretty good. Maybe a little too much detail in some cases, i.e some things feel a bit repetitive, but I think that's ok for now.

In `release-process.md`, there are still two mentions of gitian in the `Update packaging repo` section.

In `guix-attest`, can you address [this comment](https://github.com/bitcoin/bitcoin/pull/22182#issuecomment-862869410) from #22182.

I am running through doing a build using a Ubuntu:Hirsute container, using the following:
```bash
docker run -it --rm --privileged ubuntu:hirsute /bin/bash
apt update -y && apt upgrade -y && apt install curl git guix make -y

addgroup guixbuild
useradd -g guixbuild -G guixbuild -d /var/empty -s $(which nologin) -c "Guix build user" --system "guixbuilder"
guix-daemon --no-substitutes --build-users-group=guixbuild &

git clone https://github.com/bitcoin/bitcoin
cd bitcoin
HOSTS="x86_64-linux-gnu" ./contrib/guix/guix-build
```

I've also updated [my Alpine based dockerfile](https://github.com/fanquake/core-review/blob/master/guix) to use Alpine 3.14, and am currently running through a from-scratch build using that. Using a build invocation like:
`time BASE_CACHE="/guix/base_cache" SOURCES_PATH="/guix/sources" SDK_PATH="/guix/SDKs" ./contrib/guix/guix-build`.

fanquake commented:
Finished the from-scratch-build in the Alpine container:
```bash
real	210m14.468s
user	700m13.502s
sys	51m25.176s
bash-5.1# find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
ceaf2111053d48d1024e9c0c7336fc9ee6b081172eca35259b2ee9c3de2b5eec  guix-build-4129134e844f/output/aarch64-linux-gnu/SHA256SUMS.part
d564c8ce3c2132bb1304e75a49b1e9860189b98745ed35e540d7c6ef9f0e5323  guix-build-4129134e844f/output/aarch64-linux-gnu/bitcoin-4129134e844f-aarch64-linux-gnu-debug.tar.gz
8c2cc0b1046d6518b7445a19b775735b688db32b1eefc8bbcd2d3aded3937fd7  guix-build-4129134e844f/output/aarch64-linux-gnu/bitcoin-4129134e844f-aarch64-linux-gnu.tar.gz
3084de9527e95ef9bc074820e190cfacdd1962fa04cef1a79f8ccd42bab0dc6b  guix-build-4129134e844f/output/arm-linux-gnueabihf/SHA256SUMS.part
9b2d147ec30ecd1573d6cd82343cccbd35375eb5ffa269edf34b4404fca8744c  guix-build-4129134e844f/output/arm-linux-gnueabihf/bitcoin-4129134e844f-arm-linux-gnueabihf-debug.tar.gz
bd8da7121fdf38b2da885a53b10103b3a53dfef2d68e38e5ca5dde8db61af641  guix-build-4129134e844f/output/arm-linux-gnueabihf/bitcoin-4129134e844f-arm-linux-gnueabihf.tar.gz
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855  guix-build-4129134e844f/output/dist-archive/SKIPATTEST.TAG
5b1ee87a8833130eee8f7c32b0537618b90ac3db815298cc734660bd1821bcd8  guix-build-4129134e844f/output/dist-archive/bitcoin-4129134e844f.tar.gz
78d3c3a94773e3d461ee8797a6d6043c5e48c607ed8cfe5ccaf7aa7f751cd553  guix-build-4129134e844f/output/powerpc64-linux-gnu/SHA256SUMS.part
aad00c9521d461871c33d9f68a41d2a8ba9ec8e962edb93bb846d77c05b3eda1  guix-build-4129134e844f/output/powerpc64-linux-gnu/bitcoin-4129134e844f-powerpc64-linux-gnu-debug.tar.gz
1a326af20ced546fe2d03bd0d5d3b446c81c8c8dfcbde49852d39764748a0524  guix-build-4129134e844f/output/powerpc64-linux-gnu/bitcoin-4129134e844f-powerpc64-linux-gnu.tar.gz
06e409cfefc4add776479b0b70482a0ec684e991685f715a732f85516da40981  guix-build-4129134e844f/output/powerpc64le-linux-gnu/SHA256SUMS.part
237b9ee0ddb60e47b86e505011a9c17b5e898e1c5bfba59abd0e863cfcf03f5b  guix-build-4129134e844f/output/powerpc64le-linux-gnu/bitcoin-4129134e844f-powerpc64le-linux-gnu-debug.tar.gz
98de1dec19a8b6f98f045aa93bc29370b3acd5148a0acd96e9c091e6bdda1e9c  guix-build-4129134e844f/output/powerpc64le-linux-gnu/bitcoin-4129134e844f-powerpc64le-linux-gnu.tar.gz
4855cf747c23dd1d60f6b8af6933ebe464820bbc840e364d4ede5d94677cd3fe  guix-build-4129134e844f/output/riscv64-linux-gnu/SHA256SUMS.part
6e0e781bbd611ad72c085d4e00686b48dff1baf29637c481b11534f9b9bf56f2  guix-build-4129134e844f/output/riscv64-linux-gnu/bitcoin-4129134e844f-riscv64-linux-gnu-debug.tar.gz
96cff3df87b6f62542293a63d169aa29a88dc795d499168273a86c81560ccecb  guix-build-4129134e844f/output/riscv64-linux-gnu/bitcoin-4129134e844f-riscv64-linux-gnu.tar.gz
b40a1645c1601d6afbac60192789ec054ce750dc6eeb5a3f25612f4231c3d7e5  guix-build-4129134e844f/output/x86_64-apple-darwin18/SHA256SUMS.part
4dd21a03916bd42e0369e59aa0f8fb88921ed7b86fc5a1e6702389bda98a8def  guix-build-4129134e844f/output/x86_64-apple-darwin18/bitcoin-4129134e844f-osx-unsigned.dmg
ff3602163bce9934457c031831828dbd016a97d596d5aa76c42b21d9d479c2c2  guix-build-4129134e844f/output/x86_64-apple-darwin18/bitcoin-4129134e844f-osx-unsigned.tar.gz
50b1b0f6f8b8eb16d8c381fb1b66e5fc844d24f2625269f7bb28c80b20eaf125  guix-build-4129134e844f/output/x86_64-apple-darwin18/bitcoin-4129134e844f-osx64.tar.gz
af318a441f563abf1b9e1b3321b13e3a14a857216238133f9f3d923e34fc081e  guix-build-4129134e844f/output/x86_64-linux-gnu/SHA256SUMS.part
3912bb99acf83778aae12d522a0d8c567371754cb31f2f13712284fc5954a443  guix-build-4129134e844f/output/x86_64-linux-gnu/bitcoin-4129134e844f-x86_64-linux-gnu-debug.tar.gz
eb0edfa33b57ecb723b80dfc6a994af7a9cbdde72e7cf05026e98085a7dcb018  guix-build-4129134e844f/output/x86_64-linux-gnu/bitcoin-4129134e844f-x86_64-linux-gnu.tar.gz
c5cf1c6985c1c430df4839db7310ac0a46fd7da79e7de1e5cfc761523900d190  guix-build-4129134e844f/output/x86_64-w64-mingw32/SHA256SUMS.part
59721ac4abcebb51b7c011785e0701e13cb90926437ef60014078889dbec5e22  guix-build-4129134e844f/output/x86_64-w64-mingw32/bitcoin-4129134e844f-win-unsigned.tar.gz
d6f919dd523c407deaf38fc842e7a84240b64bc5c759796965b5b20618e15e2a  guix-build-4129134e844f/output/x86_64-w64-mingw32/bitcoin-4129134e844f-win64-debug.zip
96dc1c8f917b086c2a7eb84a780d4ee718eb1a608c17dfe8c29c07825f057ab6  guix-build-4129134e844f/output/x86_64-w64-mingw32/bitcoin-4129134e844f-win64-setup-unsigned.exe
e0e5925b0a898b9cc963f7a69bf9b4ada766332fffd939db142f357ee294b514  guix-build-4129134e844f/output/x86_64-w64-mingw32/bitcoin-4129134e844f-win64.zip
```

Then ran a `bash-5.1# ./contrib/guix/guix-clean` and built again, getting the same results in ~80 minutes iirc.

Unfortunately the `--no-substitutes` Hirsute build has bombed out, one of the unit tests run at the end of compiling coreutils 8.32 seems to be failing. Something to follow up on later:
```bash
================================================
   GNU coreutils 8.32: ./tests/test-suite.log
================================================

# TOTAL: 621
# PASS:  477
# SKIP:  143
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2
...
FAIL: tests/tail-2/inotify-dir-recreate
...
 inotify
-tail: 'dir/file' has become inaccessible: No such file or directory
-tail: directory containing watched file was removed
-tail: inotify cannot be used, reverting to polling
-tail: 'dir/file' has appeared;  following new file
+ fail=1
+ Exit 1
```

[Full log](https://gist.github.com/fanquake/0fa8349c0bfd9c3381de4ec434c2ddb4).

dongcarl committed:
guix: Update various check_tools lists

dongcarl committed:
guix/INSTALL: Add coreutils/inotify-dir-recreate troubleshooting

dongcarl commented:
Pushed 739556bfd4...781812a9cd
- Various fixups
- Add coreutils/inotify-dir-recreate troubleshooting information

-----

> In `release-process.md`, there are still two mentions of gitian in the `Update packaging repo` section.

Fixed!

> In `guix-attest`, can you address [this comment](https://github.com/bitcoin/bitcoin/pull/22182#issuecomment-862869410) from #22182.

Fixed!

> I am running through doing a build using a Ubuntu:Hirsute container, using the following...

Thanks for testing!

> I've also updated [my Alpine based dockerfile](https://github.com/fanquake/core-review/blob/master/guix) to use Alpine 3.14, and am currently running through a from-scratch build using that. Using a build invocation like:
> `time BASE_CACHE="/guix/base_cache" SOURCES_PATH="/guix/sources" SDK_PATH="/guix/SDKs" ./contrib/guix/guix-build`.

Great to test with a musl-based system too!

> Unfortunately the --no-substitutes Hirsute build has bombed out, one of the unit tests run at the end of compiling coreutils 8.32 seems to be failing. Something to follow up on later...

This one I've encountered before when testing Guix's v1.3.0 release. I've added some troubleshooting documentation in the last commit of the latest push: https://github.com/bitcoin/bitcoin/pull/21711/commits/781812a9cd2b85485e300db58a1ca577a0749849

MarcoFalke commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,787 @@
+
+## Optional setup
+
+At this point, you are set up to [use Guix to build bitcoin](./README.md#usage).
```
4a6d7454e8a2a666216ff0d6f7260922bb72296d:

"build Bitcoin Core"

MarcoFalke commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,787 @@
+
+For reference, the graphic below outlines Guix v1.3.0's dependency graph:
+
+![boostrap map](guix-v1.3.0-bootstrap.png)
```
Same commit: Is it really needed to add a 0.2 MB binary blob to the git tree? Is it expected that the blob is updated on every new guix version?

It could be an external link or at least optimized (still 0.1 MB, though)

```
![boostrap map](https://user-images.githubusercontent.com/6399679/125064185-a9a59880-e0b0-11eb-82c1-9b8e5dc9950d.png)

MarcoFalke replied:
(I used https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/optimize-pngs.py for this)

dongcarl replied:
I expect that this'll only have to be updated every time we do a release, and I think it'll be much less relevant once `guix` is available as a package on most stable distros.

I don't mind how it's hosted so long as people can access it. Where would be a place that makes sense?

MarcoFalke replied:
You can use above link `![boostrap map](https://user-images.githubusercontent.com/6399679/125064185-a9a59880-e0b0-11eb-82c1-9b8e5dc9950d.png)` if you don't want to upload it anywhere else.

dongcarl replied:
Done!

maflcko reviewed (approved):
Approach ACK 9f04eaac6bbf1256a5c0b1c6e818f730d85e239c

darosior reviewed (commented):
A couple of comments on `INSTALL.md` first, as it looks like managing to `guix pull` succesfully is going to take me quite some time before i get to the remaining part..

Anyways as someone who didn't follow previous Guix work it's amazing documentation! Hope it'll help to bring more builders :)

dongcarl committed:
guix/INSTALL: Guix installs init scripts in libdir

laanwj commented on doc/release-process.md:
```diff
@@ -212,93 +154,84 @@ Codesigner only: Sign the windows binaries:
+       interested in debugging can run guix to generate the files for
+       themselves. To avoid end-user confusion about which file to pick, as well
+       as save storage space *do not upload these to the bitcoincore.org server,
+       nor put them in the torrent*.
```
Agree. I think the main reason to leave `-debug` out has been because the files are huge, and due to determinism can be generated on demand anyway by anyone with enough dev expertise to need them. But not confusing users with a lot of files makes sense too.

dongcarl replied:
Yup! To be clear here, it was you who originally wrote this comment 5 years ago :-)

laanwj commented on doc/release-process.md:
```diff
@@ -212,93 +154,84 @@ Codesigner only: Sign the windows binaries:
-bitcoin-${VERSION}.tar.gz
-bitcoin-${VERSION}-win64-setup.exe
-bitcoin-${VERSION}-win64.zip
+$(cat all.SHA256SUMS)
```
Speaking of filenames, I have one manual rename left in my release script (which prepares the distribution directory for upload to bitcoincore[.]org) at the moment:
```bash
mv ${SOURCEDIR}/bitcoin-osx-signed.dmg ${DESTDIR}/bitcoin-${VERSIONTO}-osx.dmg
```
I guess this is no longer needed with the GUIX build, as the signed output will contain the version number?

dongcarl replied:
Yes I believe so!

laanwj commented on doc/release-process.md:
```diff
@@ -212,93 +154,84 @@ Codesigner only: Sign the windows binaries:
-bitcoincore.org.
+    2. The combined clear-signed message you just created `SHA256SUMS.asc`
+
+- A `.torrent` will appear in the directory after a few minutes. Optionally help
```
`bitcoincore.org` does not have functionality to automatically generate torrents. This has been done manually (I use `transmission-gtk`) for a long time.

laanwj replied:
FWIW, I use the following trackers:
```
udp://tracker.openbittorrent.com:80
udp://tracker.opentrackr.org:1337/announce
udp://tracker.coppersurfer.tk:6969/announce

udp://tracker.leechers-paradise.org:6969/announce
udp://explodie.org:6969/announce
udp://tracker.torrent.eu.org:451/announce

udp://tracker.bitcoin.sprovoost.nl:6969
```
After adding @Sjors's tracker, you need to ping him with the magnet hash so he  can whitelist it.
(I don't think these details make sense to document in the release process, it seems to be kind of transient information suspect to change, but just FYI)

dongcarl replied:
Good to have these written down even if it's in a review comment :-)
Do you think there's any torrent-related instruction that does belong in release-process.md?

laanwj replied:
Maybe that we create the torrent from the `bitcoin-core-x.x.x` directory (so the directory itself is in the torrent, with in it the archives for every platform as well as the `SHA256SUMS.asc`).
When you do this in `transmission-gtk` it makes a torrent file called `bitcoin-core-x.y.z.torrent` which I've had to rename manually to `bitcoin-x.y.z.torrent` before uploading, as that has been the naming scheme e.g. https://bitcoincore.org/bin/bitcoin-core-0.21.1/bitcoin-0.21.1.torrent.

Somewhat amusing: A commonly asked question is why the `SHA256SUMS.asc` doesn't contain a hash of the torrent, but this is because that would create a chicken-egg problem :smile:

dongcarl replied:
Added some info in 181ba751f278f5aac94c5ba1b8beeb6b3ec4629e, let me know if I'm missing anything

laanwj replied:
Looks good to me now!

laanwj commented on contrib/guix/README.md:
```diff
@@ -9,83 +9,171 @@ downloads.
 - 16GB of free disk space on the partition that /gnu/store will reside in
-- 8GB of free disk space per platform triple you're planning on building (see
-  the `HOSTS` environment variable description)
+- 8GB of free disk space **per platform triple** you're planning on building
```
Thanks for highlighting that this is per platform tuple :smile:

dongcarl commented on contrib/guix/INSTALL.md:
````diff
@@ -0,0 +1,811 @@
+Example for `systemd`, run as `root`:
+
+```sh
+# Create guix-daemon-original.service by modifying guix-daemon.service
````
Fixed in 5e23466e04e8c46aba39a1e3bf297d1947455740

dongcarl commented on contrib/guix/INSTALL.md:
````diff
@@ -0,0 +1,811 @@
+sudo --login guix pull --commit=<particular-commit>
+```
+
+`guix pull` is quite a long process (especially if you're using
````
:crossed_fingers:

dongcarl commented on contrib/guix/guix-build:
```diff
@@ -18,7 +18,7 @@ source "$(dirname "${BASH_SOURCE[0]}")/libexec/prelude.bash"
 ################
 
-check_tools cat mkdir make git guix
+check_tools cat mkdir make getent git guix
```
Fixed in fc4f8449f34e32b0b9ac9d218d6c3264b02467ba

jarolrod reviewed (commented):
Approach ACK, running through the Build from source instructions now

dongcarl committed:
doc/release-process: Add torrent creation details

Source: https://github.com/bitcoin/bitcoin/pull/21711#discussion_r668754244

dongcarl commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,816 @@
+4. Using a **distribution-maintained package** [⤓ skip to section][install-distro-pkg]
+   - Maintained by distribution's Guix package maintainer
+   - Normal difficulty (manual setup required)
+   - Works only on distributions with Guix packaged, at time of writing:
```
Right, fixed!

dongcarl commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,816 @@
+
+## Options 1 and 2: Using the official shell installer script or binary tarball
+
+The installation instructions for both the offical shell installer script and
```
😰 Fixed!

dongcarl commented on contrib/guix/INSTALL.md:
````diff
@@ -0,0 +1,816 @@
+sudo apt install guix
+```
+
+For up-to-date information on Debian and Ubuntu's release history:
````
Going to leave it there since it doesn't hurt and in case people don't know how Debian/Ubuntu versioning works ("I have bionic, but is that after hirsute or before?")

laanwj commented:
ACK 181ba751f278f5aac94c5ba1b8beeb6b3ec4629e

kallewoof commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,788 @@
+3. Using fanquake's **Docker image** [↗︎ external instructions][install-fanquake-docker]
+   - Maintained by fanquake
+   - Easy (automatically performs *some* setup)
+   - Works on most Linux distributions
```
Commit bf244f908ec92d223fc8272cda66a4dc13545d8c

Works on Mac and (I think) Windows too.

kallewoof reviewed (commented):
ACK 181ba751f278f5aac94c5ba1b8beeb6b3ec4629e

Will attempt to follow instructions on my Mac.

dongcarl commented:
Pushed 181ba751f2...fac4814106
- Addressed https://github.com/bitcoin/bitcoin/pull/21711#discussion_r670970289

fanquake reviewed (approved):
ACK fac4814106c796b8786dd90053513cc35142dfe5 - going to go ahead and merge this now. It's a lot of documentation, and could probably be nit-picked / improved further, however, that can continue over the next few weeks. I'm sure more (backportable) improvements / clarifications will be made while we progress through RCs towards a new release.

hebasto commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,813 @@
+systems to build packages from source will be helpful. *hic sunt dracones.*
+
+I strongly urge you to at least skim through the entire section once before you
+start issuing commands, as it will save you a lot of unncessary pain and
```
typo: unncessary ==> unnecessary

hebasto commented on contrib/guix/INSTALL.md:
```diff
@@ -0,0 +1,813 @@
+
+For reference, the graphic below outlines Guix v1.3.0's dependency graph:
+
+![boostrap map](https://user-images.githubusercontent.com/6399679/125064185-a9a59880-e0b0-11eb-82c1-9b8e5dc9950d.png)
```
typo: boostrap ==> bootstrap
